### PR TITLE
Topology

### PR DESCRIPTION
**Pull request by gertjanvanzwieten**

### CODE REVIEW

File: nutils/mesh.py [hunk 1]
```diff
@@ -669,15 +668,15 @@ def unitsquare(nelems, etype):
...
             topo = topology.ConnectedTopology(space, References.from_iter(references, 2), transforms, transforms, connectivity)
 
-        geom = (topo.basis('std', degree=1) * coords.T).sum(-1)
+        geom = topo.basis('std', degree=1) @ coords
         x, y = topo.boundary.sample('_centroid', None).eval(geom).T
-        bgroups = dict(left=x < .1, right=x > nelems-.1, bottom=y < .1, top=y > nelems-.1)
-        topo = topo.withboundary(**{name: topo.boundary[numpy.where(mask)[0]] for name, mask in bgroups.items()})
-        return topo, geom / nelems
+        btopo = topo.boundary
+        topo = topo.withboundary(left=btopo[x==0], right=btopo[x==nelems], bottom=btopo[y==0], top=btopo[y==nelems])
```
Comment (joostvanzwieten) [inline nutils/mesh.py:674]: Isn't the equality check (e.g. `x==0`) fragile? I suggest to move this particular change into a separate commit for bisection purposes in case things go wrong.

Comment (gertjanvanzwieten) [reply nutils/mesh.py]: fixed by reverting to 0.1 margins

**Review by joostvanzwieten — APPROVED** [review]
